File: java-instagram-web/src/Client/Component/Server/Task.php
<?php
namespace Client\Component\Server;

class Task {

    private $tags;
    private $sort;
    private $cloud;
    private $DATA_OUT_FOLDER;
    private $DATA_IN_FOLDER;
    private $MAP_FOLDER;
    private $task_id;
    private $status;
    private $result;
    private $completed_state;

    public function __construct($tags, $sort, $cloud){
        $this->tags = $tags;
        $this->sort = $sort;
        $this->cloud = $cloud;
        $this->DATA_IN_FOLDER="";
        $this->DATA_OUT_FOLDER="";
        $this->MAP_FOLDER="";
        $this->task_id=0;
        $this->status="pending";
        $this->result="";
        $this->completed_state=0;
    }

    public function getCommand(){
        return [
            'tags'=>$this->tags,
            'sort'=>$this->sort,
            'cloud'=>$this->cloud
        ];
    }



}

File: global_server/src/task/Task.java
package task;

import config.Messages;

/**
 * Task Type by PHP-Client Server
 */
public class Task {

    public String tag;
    public int sort;
    public int cloud;
    public String DATA_OUT_FOLDER;
    public String DATA_IN_FOLDER;
    public String MAP_FOLDER;
    public int task_id;
    public String status;
    public String result;
    public int completed_state;

    /**
     * Constructor
     * @param tag
     * @param sort
     * @param cloud
     */
    public Task(String tag, int sort, int cloud){
        this.tag = tag;
        this.sort = sort;
        this.cloud = cloud;
        this.status = Messages.task_pending;
        this.result = "";
        this.completed_state=0;
    }

    /**
     * Settings data input, output, map directories
     * @param DATA_IN_FOLDER
     * @param DATA_OUT_FOLDER
     * @param MAP_FOLDER
     */
    public void setDataDirectories(String DATA_IN_FOLDER, String DATA_OUT_FOLDER, String MAP_FOLDER){
        this.DATA_IN_FOLDER=DATA_IN_FOLDER;
        this.DATA_OUT_FOLDER=DATA_OUT_FOLDER;
        this.MAP_FOLDER=MAP_FOLDER;
    }

    /**
     * Settings result location
     * @param result
     */
    public void setResult(String result)
    {
        this.result = result;
    }

    /**
     * Verification that request seems normal
     * @return
     */
    public boolean analyseParameters(){
        boolean status = false;
        if(!tag.trim().equals("")){
            status=true;
        }
        return status;
    }

    @Override
    public String toString() {
        return "Task{" +
                "tag='" + tag + '\'' +
                ", sort=" + sort +
                ", cloud=" + cloud +
                ", DATA_OUT_FOLDER='" + DATA_OUT_FOLDER + '\'' +
                ", DATA_IN_FOLDER='" + DATA_IN_FOLDER + '\'' +
                ", MAP_FOLDER='" + MAP_FOLDER + '\'' +
                ", task_id=" + task_id +
                ", status='" + status + '\'' +
                ", result='" + result + '\'' +
                ", completed_state=" + completed_state +
                '}';
    }

}
